src/vs/base/parts/tree/browser/treeModel.ts

```diff
@@ -969,7 +972,17 @@ export class TreeModel {
 		let eventData: IRefreshEvent = { item: item, recursive: recursive };
 		this._onRefresh.fire(eventData);
 		return item.refresh(recursive).then(() => {
+			/**
+			 * Refresh loading status for item will be instantly lost upon refresh of item self
+			 * (usually before the spinner shows). we need to temporarily bring it back upon
+			 * actual refresh completion to let the user know the refresh happened with a short
+			 * appearance of the loading spinner.
+			 */
+			this.addTraits('loading', [element]);
 			this._onDidRefresh.fire(eventData);
+			setTimeout(function () {
```

Comment:
If you use a lambda function (`() => { ... }`) then you don't need a separate var to capture this, it will do it automatically. (i.e. you can remove the self var)